tests/cpp/test_gpu_tensorcore.cpp

```diff
@@ -1671,175 +1671,236 @@ TEST_F(GPUTTensorCoreTest, FusionAmpereViewMatmulTN_CUDA) {
-  // Schedule mma input
-  tv0cr->applyMmaSwizzle(MmaOperand::A);
+    // [M,N,K] or [M, K, N] if use_mkn_dim_order is true
+    auto tv0b = use_mkn_dim_order ? broadcast(tv0, {false, false, true})
+                                  : broadcast(tv0, {false, true, false});
+    auto tv1b = broadcast(tv1, {true, false, false});
 
-  // [... Mi, Ni, Ki]
-  tv0b->reorder({{-2, -3}, {-3, -2}});
-  tv0b->applyMmaSwizzle(MmaOperand::A);
+    // propagate the allocation domain in case there's one.
+    if (use_mkn_dim_order) {
+      tv1b->setAllocationDomain(
+          {tv1b->axis(0), tv1b->axis(2), tv1b->axis(1)}, true);
+    }
 
-  tv1cr->applyMmaSwizzle(MmaOperand::B);
-  tv1b->applyMmaSwizzle(MmaOperand::B);
+    // [M, K, N] or [M, N, K]
+    auto tv2 = use_mkn_dim_order ? fusedMultiplySum(tv0b, tv1b, {1})
+                                 : fusedMultiplySum(tv0b, tv1b, {2});
 
-  // Schedule mma output
-  tv2c->applyMmaSwizzle(MmaOperand::Accumulator);
-  tv2->applyMmaSwizzle(MmaOperand::Accumulator);
+    fusion.addOutput(tv2);
 
-  // Parallelize
-  //  0   1  2  3   4   5  6   7  8  9  10
-  // [Mo No Ko Kwo Mwo Nwo Mw Nw (Mi Ni Ki)]
-  tv2c->axis(4)->parallelize(ParallelType::TIDz);
-  tv2c->axis(5)->parallelize(ParallelType::TIDy);
+    auto mma_ops = ir_utils::getOpsOfType<MmaOp>(&fusion);
+    NVF_CHECK(
+        1 == mma_ops.size(),
+        "Invalid number of MmaOp instances in fusion definition, expected 1, got ",
+        mma_ops.size());
+    mma_ops.front()->setMacro(MmaMacro::Turing_16_8_16);
+
+    auto tv0cw = tv0->cacheAfter(LoadStoreOpType::CpAsync);
+    auto tv0cr = tv0cw->cacheAfter(LoadStoreOpType::LdMatrix);
+    auto tv1cw = tv1->cacheAfter(LoadStoreOpType::CpAsync);
+    auto tv1cr = tv1cw->cacheAfter(LoadStoreOpType::LdMatrix);
+
+    // In the case that input B has an (transposed [M, N, K] ) allocation domain
+    // and it's rfactor domain in of the order [M, K, N], reorder it to the
+    // expected [M, N, K] and propagate it backwards.
+    if (use_mkn_dim_order) {
+      // If we have inputs with rfactor domains [M, K], [K, N]
+      // first check that broadcasted B has an allocation domain and A does not.
+      NVF_CHECK(
+          tv1b->hasAllocation(),
+          "B after brodcast should have an allocation domain");
+      NVF_CHECK(
+          !tv0b->hasAllocation(),
+          "A after brodcast should not have an allocation domain");
+
+      // Get the permutation that describes the difference
+      // between the rfactor domain and allocation domain.
+      auto perm =
+          ir_utils::computePermutation(
+              tv1b->getMaybeRFactorDomain(), tv1b->getAllocationDomain())
+              .value();
+
+      NVF_CHECK(perm.size() == 3, "The permutation vector size is incorrect");
+      // We reorder from [M, K, N] -> [M, N, K], perm:[0, 2, 1]
+      NVF_CHECK(
+          perm.at(0) == 0 && perm.at(1) == 2 && perm.at(2) == 1,
+          "permutation vector is incorrect");
```

Comment:
```suggestion
  EXPECT_THAT(perm, ElementsAre(0, 2, 1));
```

I think this is easier to read. You might want to add `using testing::ElementsAre;` at the beginning if it's not already there.